server/models/ClubProfileModel.js

```diff
@@ -0,0 +1,39 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const clubProfileSchema = new Schema({
+    name: { type: String, required: true },
+    description: { type: String, required: true },
+    lastUpdated: { type: Date, default: Date.now() },
+    imageUrl: { type: String, default: '' },
+    memberRange: { type: String, required: true },
+    acceptingMembers: { type: String, required: true },
+    applicationRequired: { type: Boolean, required: true },
+    meetingFrequency: { type: String, required: true },
+    socialLinks: {
+        facebook: { type: String, default: '' },
+        email: { type: String, default: '' },
+        website: { type: String, default: '' },
+        instagram: { type: String, default: '' },
+    },
+    tags: {
+        type: [String],
+        required: true
+    },
+    highlights: {
+        type: [String],
+        default: []
+    },
+    howToJoin: { type: String, default: '' },
+    applicationLink: { type: String, default: '' },
+    similarClubs: {
+        type: [mongoose.Types.ObjectId],
+        default: []
+    },
```

Comment:
consider removing this field from the schema; instead, every time a GET on club profile is received, return the club profile itself as well as run a real-time query to find similar clubs (their indexes can be used to generate links).